apps/wpcom-block-editor/src/wpcom/features/tracking.js

```diff
@@ -414,8 +414,19 @@ const trackSaveEntityRecord = ( kind, name, record ) => {
 		name === 'wp_template_part'
 	) {
 		ignoreNextReplaceBlocksAction = true;
+		const convertedParentBlocks = select( 'core/block-editor' ).getBlocksByClientId(
+			select( 'core/block-editor' ).getSelectedBlockClientIds()
+		);
+		// We fire the event with and without the block names. We do this to
+		// make sure the event is tracked all the time. The block names
+		// might become a string that's too long and as a result it will
+		// fail because of URL length browser limitations.
+		tracksRecordEvent( 'wpcom_block_editor_convert_to_template_part', {
+			variation_slug: record.area,
+		} );
 		tracksRecordEvent( 'wpcom_block_editor_convert_to_template_part', {
 			variation_slug: record.area,
+			block_names: getFlattenedBlockNames( convertedParentBlocks ).join( ',' ),
```

Comment:
Im not in love with the idea of sending two events... but given the slack discussions that seems like it makes sense for now!

Reply:
Yeah... sending two events is a bit gross. But I'm not sure how we could limit the length of the block_names property. Since the URL length will vary based on other properties too.